netdata/netdata
# Add data query support for archived charts

##### Summary
Add the ability to query archived charts. This will only work if the dbengine memory mode is used.

After an agent restart the archived charts are not loaded in memory. When a `/api/v1/data` request is received for such a chart it will lookup the chart in the sqlite database, build the necessary chart and dimension structures and perform the query.

##### Component Name
database

##### Test Plan
- Setup a dbengine-enabled parent with a child (the parent should use dbengine to store the data at least for the child)
- Start the parent and the child e.g. **child1**
  - On the parent you can execute a query 
     `http://parent_ip:19999/host/child1/api/v1/data?chart=system.cpu&after=-10` 
     and receive the collected values for the child
- Stop the parent and child
- Start only the parent and try to execute
  `http://parent_ip:19999/host/child1/api/v1/data?chart=system.cpu&after=-10` 
   it should fail with a `Chart is not found` message

---

- Stop the parent, apply the PR and restart
- Execute
  `http://parent_ip:19999/host/child1/api/v1/data?chart=system.cpu&after=-10` 
   It should return the last collected values for the system.cpu chart for the child
